frontend/src/app/(auth)/signin/page.tsx

```diff
@@ -30,12 +30,32 @@ function Copyright(props: any) {
   );
 }
 
-export default function Login() {
+export default function SignInPage() {
 
+const makeApiCall = async (data:FormData)=>{  
```

Comment:
Hi. I noticed that API loading states are managed manually in the current implementation.  I got a recommendation for React Query - a library that simplifies data fetching and loading state management
 (https://tanstack.com/query/v3/docs/react/quick-start). 

Here's a quick example of how it can be used:

```
import { useQuery } from 'react-query';

const fetchData = async () => {
  // API call logic here
};

const MyComponent = () => {
  const { data, isLoading, error } = useQuery('myQueryKey', fetchData);

  if (isLoading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <p>Error: {error.message}</p>;
  }

  // Render your component with the fetched data
  return <div>{data}</div>;
};
```
This approach might streamline our code and enhance maintainability. What are your thoughts?

I will try to implement it in my PR and we can discuss and decide :)